utils/build-script-impl

```diff
@@ -2228,48 +2230,46 @@ for host in "${ALL_HOSTS[@]}"; do
-
-                    if [[ -d "${DEST_BUILTINS_DIR}" ]]; then
-                        for HOST_CXX_BUILTINS_PATH in "${HOST_LIB_CLANG_DIR}"/*; do
-                            HOST_CXX_BUILTINS_DIR="${HOST_CXX_BUILTINS_PATH}/lib/darwin"
-                            echo "copying compiler-rt embedded builtins from ${HOST_CXX_BUILTINS_DIR} into the local clang build directory ${DEST_BUILTINS_DIR}."
-
-                            for OS in ios watchos tvos; do
-                                LIB_NAME="libclang_rt.$OS.a"
-                                HOST_LIB_PATH="$HOST_CXX_BUILTINS_DIR/$LIB_NAME"
-                                if [[ -f "${HOST_LIB_PATH}" ]]; then
-                                    call cp "${HOST_LIB_PATH}" "${DEST_BUILTINS_DIR}/${LIB_NAME}"
-                                elif [[ "${VERBOSE_BUILD}" ]]; then
-                                    echo "no file exists at ${HOST_LIB_PATH}"
-                                fi
-                            done
+        call "${CMAKE_BUILD[@]}" "${build_dir}" $(cmake_config_opt ${product}) -- "${BUILD_ARGS[@]}" ${build_targets[@]}
+
+        # When we are building LLVM copy over the compiler-rt
+        # builtins for iOS/tvOS/watchOS to ensure that Swift's
+        # stdlib can use compiler-rt builtins when targetting iOS/tvOS/watchOS.
+        if [[ "${product}" = "llvm" ]] && [[ "${BUILD_LLVM}" = "1" ]] && [[ "$(uname -s)" = "Darwin" ]]; then
+            HOST_CXX_DIR=$(dirname "${HOST_CXX}")
+            HOST_LIB_CLANG_DIR="${HOST_CXX_DIR}/../lib/clang"
+            DEST_LIB_CLANG_DIR="$(build_directory_bin ${host} llvm)/../lib/clang"
+
+            if [[ -d "${HOST_LIB_CLANG_DIR}" ]] && [[ -d "${DEST_LIB_CLANG_DIR}" ]]; then
+                DEST_CXX_BUILTINS_VERSION=$(ls "${DEST_LIB_CLANG_DIR}" | awk '{print $0}')
+                DEST_BUILTINS_DIR="$(build_directory_bin ${host} llvm)/../lib/clang/$DEST_CXX_BUILTINS_VERSION/lib/darwin"
+
+                if [[ -d "${DEST_BUILTINS_DIR}" ]]; then
+                    for HOST_CXX_BUILTINS_PATH in "${HOST_LIB_CLANG_DIR}"/*; do
+                        HOST_CXX_BUILTINS_DIR="${HOST_CXX_BUILTINS_PATH}/lib/darwin"
+                        echo "copying compiler-rt embedded builtins from ${HOST_CXX_BUILTINS_DIR} into the local clang build directory ${DEST_BUILTINS_DIR}."
+
+                        for OS in ios watchos tvos; do
+                            LIB_NAME="libclang_rt.$OS.a"
+                            HOST_LIB_PATH="$HOST_CXX_BUILTINS_DIR/$LIB_NAME"
+                            if [[ -f "${HOST_LIB_PATH}" ]]; then
+                                call cp "${HOST_LIB_PATH}" "${DEST_BUILTINS_DIR}/${LIB_NAME}"
+                            elif [[ "${VERBOSE_BUILD}" ]]; then
+                                echo "no file exists at ${HOST_LIB_PATH}"
+                            fi
                         done
-                    fi
+                    done
```

Comment:
This whole thing is kinda ugly but I just removed the outer `if` condition that use the `skip_build` variable.

Reply:
Yeah, its all whitespace diff really.  Hint: add `?w=1` to the diff URL to get the real diff